modules/cli/src/main/scala/scala/cli/commands/scalafix/Scalafix.scala

```diff
@@ -0,0 +1,197 @@
+      case v if v.startsWith("2.13.") => "2.13"
+      case v if v.startsWith("3.")    => "2.13"
+      case _ =>
+        logger.error("Unsupported scala version " + scalaVersion)
+        sys.exit(1)
+    val configFilePathOpt = options.scalafixConf.map(os.Path(_, os.pwd))
+    val relPaths          = sourcePaths.map(_.toNIO.getFileName)
+
+    val scalafix = ScalafixInterface
+      .fetchAndClassloadInstance(scalaBinaryVersion)
+      .newArguments()
+      .withWorkingDirectory(workspace.toNIO)
+      .withPaths(relPaths.asJava)
+      .withRules(options.rules.asJava)
+      .withConfig(configFilePathOpt.map(_.toNIO).toJava)
+      .withScalaVersion(scalaVersion)
+
+    logger.debug(
+      s"Processing ${sourcePaths.size} Scala sources"
+    )
+
+    val rulesThatWillRun: Either[ScalafixException, mutable.Buffer[ScalafixRule]] =
+      try
+        Right(scalafix.rulesThatWillRun().asScala)
+      catch
+        case e: ScalafixException => Left(e)
+    val needToBuild: Boolean = rulesThatWillRun match
+      case Right(rules) => rules.exists(_.kind().isSemantic)
+      case Left(_)      => true
+
+    val preparedScalafixInstance = if (needToBuild) {
+      val res = Build.build(
+        inputs,
+        buildOptionsWithSemanticDb,
+        compilerMaker,
+        None,
+        logger,
+        crossBuilds = false,
+        buildTests = false,
+        partial = None,
+        actionableDiagnostics = actionableDiagnostics
+      )
+      val builds = res.orExit(logger)
+
+      val successfulBuildOpt = for {
+        build <- builds.get(Scope.Main)
+        sOpt  <- build.successfulOpt
+      } yield sOpt
+
+      val classPaths = successfulBuildOpt.map(_.fullClassPath).getOrElse(Seq.empty)
+      val externalDeps =
+        options.shared.dependencies.compileOnlyDependency ++ successfulBuildOpt.map(
+          _.options.classPathOptions.extraCompileOnlyDependencies.values.flatten.map(_.value.render)
+        ).getOrElse(Seq.empty)
+      val scalacOptions = options.shared.scalac.scalacOption ++ successfulBuildOpt.map(
+        _.options.scalaOptions.scalacOptions.toSeq.map(_.value.value)
+      ).getOrElse(Seq.empty)
+
+      scalafix
+        .withScalacOptions(scalacOptions.asJava)
+        .withClasspath(classPaths.map(_.toNIO).asJava)
+        .withToolClasspath(Seq.empty.asJava, externalDeps.asJava)
+    }
+    else
+      scalafix
+
+    val customScalafixInstance = preparedScalafixInstance
+      .withParsedArguments(options.scalafixArg.asJava)
+
+    val errors = if (options.check) {
+      val evaluation = customScalafixInstance.evaluate()
+      if (evaluation.isSuccessful)
+        evaluation.getFileEvaluations.foldLeft(List.empty[String]) {
+          case (errors, fileEvaluation) =>
+            val problemMessage = fileEvaluation.getErrorMessage.toScala.orElse(
+              fileEvaluation.previewPatchesAsUnifiedDiff.toScala
+            )
+            errors ++ problemMessage
+        }
```

Comment:
With this implementation, `--check` effectively ignores diagnostics (linters)

```
➜  /tmp cat Test.scala
package foo

final object Hello {
  def main(args: Array[String]): Unit = {
    val name = null
  }
}
```

Expected (no `--check`)
```
➜  /tmp /Users/brice.jaglin/git/scala/scala-cli/out/cli/3.3.3/launcher.dest/launcher --power scalafix --rules=DisableSyntax --scalafix-arg=--settings.DisableSyntax.noNulls=true Test.scala && echo OK
Some utilized features are marked as experimental:
 - `scalafix` sub-command
 - `--rules=DisableSyntax` option
 - `--scalafix-arg=--settings.DisableSyntax.noNulls=true` option
Please bear in mind that non-ideal user experience should be expected.
If you encounter any bugs or have feedback to share, make sure to reach out to the maintenance team at https://github.com/VirtusLab/scala-cli
/private/tmp/Test.scala:5:16: error: [DisableSyntax.null] null should be avoided, consider using Option instead
    val name = null
               ^^^^
A Scalafix linter error was reported
```

Not expected (with `--check`)
```
➜  /tmp /Users/brice.jaglin/git/scala/scala-cli/out/cli/3.3.3/launcher.dest/launcher --power scalafix --rules=DisableSyntax --scalafix-arg=--settings.DisableSyntax.noNulls=true Test.scala --check && echo OK
Some utilized features are marked as experimental:
 - `scalafix` sub-command
 - `--rules=DisableSyntax` option
 - `--scalafix-arg=--settings.DisableSyntax.noNulls=true` option
 - `--check` option
Please bear in mind that non-ideal user experience should be expected.
If you encounter any bugs or have feedback to share, make sure to reach out to the maintenance team at https://github.com/VirtusLab/scala-cli
OK
```


Instead of evaluating manually and reconstructing logs from lints/rewrites in each file (which might result in missing upcoming features), you should call `run()` with the `--check` flag and redirect the ScalafixMainCallback to the scala-cli logger. That's [what sbt-scalafix does](https://github.com/scalacenter/sbt-scalafix/blob/main/src/main/scala/scalafix/internal/sbt/ScalafixLogger.scala).

Reply:
This shows that invocation output (either with or without --check for a linter rule such as DisableSyntax, and with --check for a rewrite rule such as RedundantSyntax) should be asserted in the tests.

Reply:
tested this loccaly - now it works as expected